dockers-apt-repo.sh

```diff
@@ -1,17 +1,40 @@
-# this is just notes going to automate!
-# Add Docker's official GPG key:
-sudo apt-get update
-sudo apt-get install ca-certificates curl
-sudo install -m 0755 -d /etc/apt/keyrings
-sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
-sudo chmod a+r /etc/apt/keyrings/docker.asc
 
-# Add the repository to Apt sources:
+# Creating encryption certificates
+if (  which ca-certificates curl )
+then
+  echo " ca-certificates curl already exist "
+else 
+  echo " Creating ca-certificates "
+  sudo apt-get update
+  sudo apt-get install ca-certificates curl
+fi
+
+# Adding and configure Docker's official GPG key
+if (  which gpg )
+then
+  echo " gpg key already exist "
+else 
+  echo " Installing the keyring program "
+  sudo install -m 0755 -d /etc/apt/keyrings 
+  echo " Configuring gpg keyring "
+  sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
+  echo " Modifying gpg file permissions "
+  sudo chmod a+r /etc/apt/keyrings/docker.asc
+fi
+
+# Add the repository to Apt sources
+# Not sure how to put this in a script
 echo \
-  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
+  "deb [GNU=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
   $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
   sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
 sudo apt-get update
 
 # Then install docker's latest
-` sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`
\ No newline at end of file
+if ( which docker )
+then 
+  echo "Dockers latest already installed"
+else
+  echo "Installing docker"
+  sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```

Comment:
You can just use apt install instead of apt-get, look up the diff between the two.

Reply:
Break this into chunks. Take what you have gotten from this site and decide what are your dependencies, what directories and permissions need to exist for you to get the outcome you are looking for? Then break all those into individual conditionals.

It's ok to have all the docker deps in one conditional in my opinion. Also do any of these installs need a `-y`?

Reply:
Next up, write a script to connect to you pi, then run this script in the pi.